pkg/signal/signal.go

```diff
@@ -0,0 +1,153 @@
+// Copyright 2019 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package signal
+
+import (
+	"bytes"
+	"encoding/binary"
+	"fmt"
+	"runtime"
+	"sync"
+
+	"github.com/cilium/cilium/pkg/bpf"
+	"github.com/cilium/cilium/pkg/byteorder"
+	"github.com/cilium/cilium/pkg/logging"
+	"github.com/cilium/cilium/pkg/logging/logfields"
+)
+
+const (
+	// SignalMapName is the BPF map name
+	SignalMapName = "cilium_signals"
+)
+
+const (
+	// SignalNatFillUp denotes potential congestion on the NAT table
+	SignalNatFillUp = iota
+	SignalTypeMax
+)
+
+const (
+	// SignalNatV4 denotes NAT IPv4 table
+	SignalNatV4 = iota
+	// SignalNatV6 denotes NAT IPv6 table
+	SignalNatV6
+)
+
+var (
+	log = logging.DefaultLogger.WithField(logfields.LogSubsys, "signal")
+
+	config = bpf.PerfEventConfig{
+		MapName:      SignalMapName,
+		Type:         bpf.PERF_TYPE_SOFTWARE,
+		Config:       bpf.PERF_COUNT_SW_BPF_OUTPUT,
+		SampleType:   bpf.PERF_SAMPLE_RAW,
+		NumPages:     2,
```

Comment:
Is this one page for the perf header and one page for data? Or actually two pages for data?